src/codeflare_sdk/cluster/cluster.py

```diff
@@ -268,19 +268,27 @@ def wait_ready(self, timeout: Optional[int] = None):
         dashboard_ready = False
         status = None
         time = 0
-        while not ready or not dashboard_ready:
+        while not ready:
             status, ready = self.status(print_to_console=False)
-            dashboard_ready = self.is_dashboard_ready()
             if status == CodeFlareClusterStatus.UNKNOWN:
                 print(
                     "WARNING: Current cluster status is unknown, have you run cluster.up yet?"
                 )
-            if not ready or not dashboard_ready:
+            if not ready:
+                if timeout and time >= timeout:
+                    raise TimeoutError(f"wait() timed out after waiting {timeout}s for cluster to be ready")
+                sleep(5)
+                time += 5
+        print("Requested cluster is up and running!")
+
+        while not dashboard_ready:
+            dashboard_ready = self.is_dashboard_ready()
+            if not dashboard_ready:
                 if timeout and time >= timeout:
-                    raise TimeoutError(f"wait() timed out after waiting {timeout}s")
+                    raise TimeoutError(f"wait() timed out after waiting {timeout}s for dashboard to be ready")
                 sleep(5)
                 time += 5
-        print("Requested cluster and dashboard are up and running!")
+        print("Dashboard is ready!")
```

Comment:
Nice catch, thanks! I used `if dashboard_ready` instead as it seemed more appropriate because it will never be true if `dashboard_check` is false. What do you think?

Reply:
Yup, that works!